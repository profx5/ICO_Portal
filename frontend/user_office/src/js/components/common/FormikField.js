import React from 'react';
import styled from 'styled-components';
import Cleave from 'cleave.js/react';

import { Field } from 'formik';
import ErrorMessage from './ErrorMessage';


const FormikField = ({errors, touched, labelText, placeholder, options, name, value, disabled, required, onChangeHandler}) => {

    return (
        <Wrapper>
            <StyledLabel htmlFor={name}>{labelText}{required && <span> *</span>}</StyledLabel>
            <Field
                name={name}
                render={({field}) => (
                    <StyledCleave className={(errors[name] && touched[name]) && 'isInvalid'} id={name} {...field} placeholder={placeholder} type="text" options={options || {delimiter: ''}}/>
                )}
            />
            {errors[name] && touched[name] && <ErrorMessage text={errors[name]}/>}
        </Wrapper>
    );
}


export default FormikField;

const Wrapper = styled.div`
    position: relative;
    height: 100%;
    label + div {
       height: 100%;
       .react-datepicker-wrapper, .react-datepicker__input-container {
            display: block;
            height: 100%;
       }
       input {
            color: #233539;
            font-size: 16px;
            font-weight: 600;
            padding: 0 20px;
            display: block;
            height: 100%;
            width: 100%;
            background: #ffffff;
            border: 1px solid #d6dfe6;
            border-radius: 2px;
            &::-webkit-input-placeholder {
                color: rgba(10,10,10,.4);
            }
       }
    }
    .DayPickerInput {
        display: block;
        height: 100%;
        input {
            color: #233539;
            font-size: 16px;
            font-weight: 600;
            padding: 0 20px;
            display: block;
            height: 100%;
            width: 100%;
            background: #ffffff;
            border: 1px solid #d6dfe6;
            border-radius: 2px;
            &::-webkit-input-placeholder {
                color: rgba(10,10,10,.4);
            }
        }
    }
`;

const StyledLabel = styled.label`
    color: #0a0a0a;
    font-size: 16px;
    display: block;
    margin-bottom: 13px;
    span {
        color: red;
    }
`;

const StyledCleave = styled(Cleave)`
    color: #233539;
    font-size: 16px;
    font-weight: 600;
    padding: 0 20px;
    display: block;
    height: 100%;
    width: 100%;
    background: #ffffff;
    border: 1px solid #d6dfe6;
    border-radius: 2px;
    &:read-only {
        color: rgba(35,53,57,.3);
    }
    &::-webkit-input-placeholder {
        color: rgba(10,10,10,.4);
    }
    &.isInvalid {
        border-color: rgb(242, 109, 109);
    }
`;
