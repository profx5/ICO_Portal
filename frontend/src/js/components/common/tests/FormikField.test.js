import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FormikField from 'js/components/common/FormikField';


configure({ adapter: new Adapter() });

it('should render stateless FormikField', () => {
    const component = shallow(<FormikField/>);
    expect(component).toMatchSnapshot();
})
