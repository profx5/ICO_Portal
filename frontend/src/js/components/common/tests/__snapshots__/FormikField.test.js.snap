// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless FormikField 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormikField />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef />,
        <FormikConnect(FieldInner)
          render={[Function]}
        />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
          ],
          "htmlFor": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    color: #0a0a0a;
    font-size: 16px;
    display: block;
    margin-bottom: 13px;
    white-space: nowrap;
    ",
              "@media (max-width: 767px)",
              " {
        font-size: 12px;
    }
    span {
        color: red;
    }
",
            ],
          },
          "displayName": "styled.label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": undefined,
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
    position: relative;
    height: 100%;
    label + div {
       height: 100%;
       input {
            color: #233539;
            font-size: 16px;
            font-weight: 500;
            padding: 0 20px;
            display: block;
            height: 100%;
            width: 100%;
            background: #ffffff;
            border: 1px solid #d6dfe6;
            border-radius: 2px;
            &::-webkit-input-placeholder {
                color: rgba(10,10,10,.4);
            }
       }
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef />,
          <FormikConnect(FieldInner)
            render={[Function]}
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "htmlFor": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    color: #0a0a0a;
    font-size: 16px;
    display: block;
    margin-bottom: 13px;
    white-space: nowrap;
    ",
                "@media (max-width: 767px)",
                " {
        font-size: 12px;
    }
    span {
        color: red;
    }
",
              ],
            },
            "displayName": "styled.label",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "label",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": undefined,
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
    position: relative;
    height: 100%;
    label + div {
       height: 100%;
       input {
            color: #233539;
            font-size: 16px;
            font-weight: 500;
            padding: 0 20px;
            display: block;
            height: 100%;
            width: 100%;
            background: #ffffff;
            border: 1px solid #d6dfe6;
            border-radius: 2px;
            &::-webkit-input-placeholder {
                color: rgba(10,10,10,.4);
            }
       }
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
